src/IB/IBFESurfaceMethod.cpp

```diff
@@ -672,19 +1037,274 @@ IBFESurfaceMethod::interpolateVelocity(const int u_data_idx,
             // that are within the patch interior.
             const Box<NDIM>& interp_box = patch->getBox();
             Pointer<PatchData<NDIM> > u_data = patch->getPatchData(u_data_idx);
+
+            const Box<NDIM> ghost_box = Box<NDIM>::grow(patch->getBox(), IntVector<NDIM>(u_ghost_num));
+
             Pointer<CellData<NDIM, double> > u_cc_data = u_data;
             if (u_cc_data)
             {
                 LEInteractor::interpolate(
                     U_qp, NDIM, x_qp, NDIM, u_cc_data, patch, interp_box, d_default_interp_spec.kernel_fcn);
             }
             Pointer<SideData<NDIM, double> > u_sc_data = u_data;
-            if (u_sc_data)
+            if (u_sc_data && !d_use_velocity_jump_conditions)
             {
                 LEInteractor::interpolate(
                     U_qp, NDIM, x_qp, NDIM, u_sc_data, patch, interp_box, d_default_interp_spec.kernel_fcn);
             }
+            else if (u_sc_data && d_use_velocity_jump_conditions)
+            {
+                LEInteractor::interpolate(
+                      U_o_qp, NDIM, x_o_qp, NDIM, u_sc_data, patch, ghost_box, d_default_interp_spec.kernel_fcn);
+
+                LEInteractor::interpolate(
+                    U_i_qp, NDIM, x_i_qp, NDIM, u_sc_data, patch, ghost_box, d_default_interp_spec.kernel_fcn);
+
+                const IntVector<NDIM>& u_gcw = u_sc_data->getGhostCellWidth();
+                const int u_depth = u_sc_data->getDepth();
+                TBOX_ASSERT(u_depth == 1);
+
+                // Keep the quadrature points that are inside the interpolation box.
+                std::vector<int> local_indices;
+                for (unsigned int k = 0; k < n_qpoints_patch; ++k)
+                {
+                    const double* const x = &x_qp[NDIM * k];
+                    const Index<NDIM> i = IndexUtilities::getCellIndex(x, patch_geom, patch_box);
+                    if (interp_box.contains(i)) local_indices.push_back(k);
+
+                    const double* const xo = &x_o_qp[NDIM * k];
+                    const Index<NDIM> io = IndexUtilities::getCellIndex(
+                        xo, x_lower_ghost, x_upper_ghost, patch_geom->getDx(), ghost_box.lower(), ghost_box.upper());
+
+                    // Some kind of assertation can be applied here using the indices of the cells away from the interfce
+
+                }
+                if (local_indices.empty()) continue;
+                Index<NDIM> ic_lower, ic_upper, ic_center;
+                boost::array<boost::array<double, 2>, NDIM> w, wr;
+#if (NDIM == 2)
+                boost::array<double, NDIM> LL, LU, UL, UU;
```

Comment:
It might clean up some code lower down to combine these into a single `boost::multi_array` that is indexed for L/U, e.g. using L=`0` and U=`1`.

Reply:
Okay I can do this.